src/_pytest/config/__init__.py

```diff
@@ -383,6 +384,7 @@ def _try_load_conftest(self, anchor):
                 if x.check(dir=1):
                     self._getconftestmodules(x)
 
+    @functools.lru_cache()
     def _getconftestmodules(self, path):
         if self._noconftest:
             return []
```

Comment:
This will be done via https://github.com/pytest-dev/pytest/commit/1acb18b8fbdf4a06577cacf4030d5725dd57aaf4 already, please take it out here.

Reply:
Oh, I forgot about that. Will do.

Reply:
Hmm.. what's with max_size=None though? Seems a bit excessive?

Reply:
> Hmm.. what's with max_size=None though? Seems a bit excessive?

The default is 128, which is a bit small IMHO, and I thought it should be used for everything - just like it was done with the previous manual caching.
If limiting this is should be probably in the range of 10k, but then it might not help with bigger test suites etc.

Reply:
I have a big test suite at work with ~5k tests that takes 25s to collect without that lru_cache and 14s with. But there is no discernible difference between infinite cache size and 128. Maybe just because the import hook is so terribly slow in comparison, but wanted to give this perspective. We might be trading more memory for no speed improvement...

Reply:
(and as an aside pytest 3.6.0 collects that suite in 9s)

Reply:
Ah, I see.
It is likely that the cache is only used for a short period (i.e. during collection of files in a directory etc).
A limit makes sense then indeed.

Reply:
Amended, 371ed5b.

#4249 might turn out to be useful there in the future.

Reply:
Amended further - 94a6396, a single lru_cache appears to be enough here.